src/finch/compiled.py

```diff
@@ -4,31 +4,54 @@
 from .tensor import Tensor
 
 
-def compiled(func):
-    @wraps(func)
-    def wrapper_func(*args, **kwargs):
-        new_args = []
-        for arg in args:
-            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
-                new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
-            else:
-                new_args.append(arg)
-
-        result = func(*new_args, **kwargs)
-        result_tensor = Tensor(jl.Finch.compute(result._obj))
-
-        return result_tensor
-
-    return wrapper_func
-
+def compiled(opt=""):
+    def inner(func):  
+        @wraps(func)  
+        def wrapper_func(*args, **kwargs):  
+            new_args = []  
+            for arg in args:  
+                if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):  
+                    new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))  
+                else:  
+                    new_args.append(arg)  
+            result = func(*new_args, **kwargs)  
+            kwargs = {"ctx": get_scheduler(name=opt)} if opt != "" else {}  
+            result_tensor = Tensor(jl.Finch.compute(result._obj, **kwargs))  
+            return result_tensor  
+        return wrapper_func  
+
+    return inner  
 
 def lazy(tensor: Tensor):
     if tensor.is_computed():
         return Tensor(jl.Finch.LazyTensor(tensor._obj))
     return tensor
 
+class AbstractScheduler():
+    pass
+
+class GalleyScheduler(AbstractScheduler):
+    def __init__(self, verbose=False):
+        self.verbose=verbose
+
+class DefaultScheduler(AbstractScheduler):
+    def __init__(self, verbose=False):
+        self.verbose=verbose
```

Comment:
let's put get_julia_scheduler inside the classes, so that we can pass different flags. Then we can just call opt.get_julia_scheduler() in compute.